aten/src/ATen/core/function_schema.h

```diff
@@ -243,7 +243,49 @@ inline bool operator!=(const FunctionSchema& lhs, const FunctionSchema& rhs) {
 
 // for debugging, make sure we can describe the call site
 inline std::ostream& operator<<(std::ostream& out, const Argument& arg) {
-  return out << arg.type()->str() << " " << arg.name() << (arg.default_value() ? "=<default>" : "");
+  bool optional_type = arg.type()->isSubclass(TypeKind::OptionalType);
+  // for adjusting the ? position.
+  // in schema, we have Tensor?(a!) input, and t(a!)?.
+  // however, t?(a!) doesn't work with schema parser.
+  // so we always use Type(alias)? format
+  std::stringstream oss;
+  if (arg.type()->isSubclass(TypeKind::ListType) && arg.N()) {
+    oss << std::static_pointer_cast<const ListType>(arg.type())->
+      getElementType()->str();
+    oss << "[" << arg.N().value() << "]";
+  } else {
+    oss << arg.type()->str();
+  }
+  if (optional_type) {
+    oss.seekp(oss.str().size() - 1);
+  }
+  if (arg.alias_info()) {
+    oss << arg.alias_info().value();
+  }
+  if (optional_type) {
+    oss << "?";
+  }
+  out << oss.str();
+  if (!arg.name().empty()) {
+    out << " " << arg.name();
+  }
+  if (arg.default_value()) {
+    out << "=";
+    //out << expand_escapes(arg.default_value().value().toStringRef());
+    if (arg.type()->kind() == c10::TypeKind::StringType) {
+        out << "\'";
```

Comment:
shouldn't it be " not ' ? or which syntax do we use?

Reply:
both work in this case, check https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/script/lexer.h#L200

Also, ''' and """ will work too.

I chose ' since i saw examples here: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/register_string_ops.cpp#L455

Just to be consistent.